build-live.sh

```diff
@@ -0,0 +1,304 @@
+  gir1.2-timezonemap-1.0 \
+  gir1.2-xkl-1.0 \
+  gparted \
+  iw \
+  keyutils \
+  kpartx \
+  kpartx-boot \
+  language-pack-en \
+  language-pack-en-base \
+  libatkmm-1.6-1 \
+  libcairomm-1.0-1 \
+  libcryptsetup4 \
+  libdebian-installer4 \
+  libdevmapper-event1.02.1 \
+  libdmraid1.0.0.rc16 \
+  libecryptfs0 \
+  libglibmm-2.4-1c2a \
+  libgtkmm-2.4-1c2a \
+  libnss3-1d \
+  libpangomm-1.4-1 \
+  libsigc++-2.0-0c2a \
+  libtimezonemap1 \
+  lvm2 \
+  pptp-linux \
+  python3-cairo \
+  python3-gi-cairo \
+  python3-icu \
+  python3-pam \
+  rdate \
+  sbsigntool \
+  ubiquity \
+  ubiquity-casper \
+  ubiquity-frontend-gtk \
+  ubiquity-ubuntu-artwork \
+  watershed \
+  wireless-regdb \
+  xserver-xorg \
+  gnome-themes-standard \
+  lightdm
+
+lxc exec $CONTAINER -- rm /usr/sbin/policy-rc.d
+
+echo "Removing diversions..."
+lxc exec $CONTAINER -- cp /bin/running-in-container.distrib /bin/running-in-container
+lxc exec $CONTAINER -- cp /sbin/initctl.distrib /sbin/initctl
+
+lxc exec $CONTAINER -- dpkg-divert --remove /bin/running-in-container
+lxc exec $CONTAINER -- dpkg-divert --remove /sbin/initctl
+
+# clean up
+
+lxc exec $CONTAINER -- apt-get clean
+lxc exec $CONTAINER -- rm /var/lib/dbus/machine-id
+lxc exec $CONTAINER -- rm /etc/init/console.override
+lxc exec $CONTAINER -- rm /etc/init/tty1.override
+lxc exec $CONTAINER -- rm /etc/init/tty2.override
+lxc exec $CONTAINER -- rm /etc/init/tty3.override
+lxc exec $CONTAINER -- rm /etc/init/tty4.override
+# this will be recreated
+lxc exec $CONTAINER -- rm /etc/network/interfaces.d/eth0.cfg
+
+kversion=$(lxc exec $CONTAINER -- ls /boot/ | grep vmlinuz | grep -v efi | tail -1 | sed s@vmlinuz-@@)
+
+lxc file pull $CONTAINER/boot/vmlinuz-${kversion} ${CD}/${FS_DIR}/vmlinuz
+lxc file pull $CONTAINER/boot/initrd.img-${kversion} ${CD}/${FS_DIR}/initrd.img
+
+lxc exec $CONTAINER -- dpkg-query -W --showformat='${Package} ${Version}\n' | sort > ${CD}/${FS_DIR}/filesystem.manifest
+
+comm -2 -3 ${CD}/${FS_DIR}/filesystem.manifest ${CD}/${FS_DIR}/filesystem-desktop.manifest > ${CD}/${FS_DIR}/filesystem-remove.manifest
+
+lxc stop $CONTAINER
+
+cat > ${CD}/boot/grub/grub.cfg << EOF
+if loadfont /boot/grub/font.pf2 ; then
+    set gfxmode=auto
+    insmod efi_gop
+    insmod efi_uga
+    insmod gfxterm
+    terminal_output gfxterm
+fi
+
+set default="0"
+set timeout=10
+
+menuentry "Automated Install" {
+  linux /casper/vmlinuz boot=casper only-ubiquity automatic-ubiquity noprompt quiet splash file=/cdrom/preseed/posm-full.seed ip=frommedia
+  initrd /casper/initrd.img
+}
+
+menuentry "Install" {
+  linux /casper/vmlinuz boot=casper only-ubiquity noprompt quiet splash file=/cdrom/preseed/posm.seed ip=frommedia
+  initrd /casper/initrd.img
+}
+EOF
+
+# export from LXC in order to get correct UID/GIDs
+lxc publish $CONTAINER --alias $CONTAINER
+lxc image export $CONTAINER ${CONTAINER}.tar.gz
+lxc image delete $CONTAINER
+sudo rm -rf rootfs/
+sudo tar zxf ${CONTAINER}.tar.gz rootfs/
```

Comment:
Per http://unix.stackexchange.com/questions/143961/unpack-modify-and-pack-initrd-as-a-user/144236#144236, it may be possible to bypass the `root` requirement.

Check the `filesystem.squashfs` included on an Ubuntu Desktop live image to see if it contains `/dev`.